frontend/src/scenes/authentication/RegionSelect.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { useValues, path, kea, actions, useActions, listeners } from 'kea'
+import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
+import { LemonSelect } from '@posthog/lemon-ui'
+import { Region } from '~/types'
+import { CLOUD_EU_URL, CLOUD_US_URL } from 'lib/constants'
+import { router } from 'kea-router'
+
+import type { regionSelectLogicType } from './RegionSelectType'
+import { PureField } from 'lib/forms/Field'
+
+export const regionSelectLogic = kea<regionSelectLogicType>([
+    path(['scenes', 'authentication', 'RegionSelect']),
+    actions({
+        onRegionChange: (region: Region | null) => ({ region }),
+    }),
+    listeners({
+        onRegionChange: ({ region }) => {
+            if (!region) {
+                return
+            }
+            const { pathname } = router.values.currentLocation
+            const newUrl = region === Region.EU ? `${CLOUD_EU_URL}${pathname}` : `${CLOUD_US_URL}${pathname}`
```

Comment:
Could also include `search` and `hash` just in case (we might want to preserve UTM params or something like that)